TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/transform/JoltJsonTransformBuilder.java

```diff
@@ -68,15 +68,29 @@ private Map<String, Object> parseSpecOperationFromResource(String resource) {
     }
 
     private Map<String, Object> getHostSwitchOperation(String targetClusterHostname) {
-        var jsonHostSwitchTemplateJson = parseSpecOperationFromResource(RESOURCE_HOST_SWITCH_OPERATION);
-        var specJson = (Map<String, Object>) jsonHostSwitchTemplateJson.get("spec");
+        var joltTransformTemplate = parseSpecOperationFromResource(RESOURCE_HOST_SWITCH_OPERATION);
+        var specJson = (Map<String, Object>) joltTransformTemplate.get("spec");
         var headersSpecJson = (Map<String, Object>) specJson.get("headers");
         headersSpecJson.put("host", targetClusterHostname);
-        return jsonHostSwitchTemplateJson;
+        return joltTransformTemplate;
     }
 
+
+    public Map<String, Object> getAddAuthorizationOperation(String authorizationHeader) {
+        var joltTransformTemplate = parseSpecOperationFromResource(ADD_AUTHORIZATION_HEADER_OPERATION);
+        var specJson = (Map<String, Object>) joltTransformTemplate.get("spec");
+        var headersSpecJson = (Map<String, Object>) specJson.get("headers");
+        headersSpecJson.put("authorization", authorizationHeader);
```

Comment:
will this overwrite an auth header if it's already there?

Reply:
Yes